Lib/tkinter/colorchooser.py

```diff
@@ -8,55 +8,67 @@
 # fixed initialcolor handling in August 1998
 #
 
-#
-# options (all have default values):
-#
-# - initialcolor: color to mark as selected when dialog is displayed
-#   (given as an RGB triplet or a Tk color string)
-#
-# - parent: which window to place the dialog on top of
-#
-# - title: dialog title
-#
 
 from tkinter.commondialog import Dialog
 
 
-#
-# color chooser class
-
 class Chooser(Dialog):
-    "Ask for a color"
+    """Create a dialog for the tk_chooseColor command.
+
+    Args:
+        master: The master widget for this dialog.  If not provided,
+            defaults to options['parent'] (if defined).
+        options: Dictionary of options for the tk_chooseColor call.
+            initialcolor: Specifies the selected color when the
+                dialog is first displayed.  This can be a tk color
+                string or a 3-tuple of ints in the range (0, 255)
+                for an RGB triplet.
+            parent: The parent window of the color dialog.  The
+                color dialog is displayed on top of this.
+            title: A string for the title of the dialog box.
+    """
 
     command = "tk_chooseColor"
 
     def _fixoptions(self):
+        """Ensure initialcolor is a tk color string.
+
+        Convert initialcolor from a RGB triplet to a color string.
+        """
         try:
-            # make sure initialcolor is a tk color string
             color = self.options["initialcolor"]
             if isinstance(color, tuple):
-                # assume an RGB triplet
+                # Assume an RGB triplet.
                 self.options["initialcolor"] = "#%02x%02x%02x" % color
         except KeyError:
             pass
 
     def _fixresult(self, widget, result):
-        # result can be somethings: an empty tuple, an empty string or
-        # a Tcl_Obj, so this somewhat weird check handles that
+        """Adjust result returned from call to tk_chooseColor.
+
+        Return both an RGB tuple of ints in the range (0, 255) and the
+        tk color string in the form #xxxxxx.
+        """
+        # Result can be many things: an empty tuple, an empty string, or
+        # a Tcl_Obj, so this somewhat weird check handles that.
```

Comment:
The input to a Python function must be a Python object, not a Tcl_Obj.  Adding "print(type(result), result)" here shows that the passed-in result is '' if [Cancel] or [X] is clicked else '#rrggbb' if [OK] is clicked.  (The default initial color is '#a0a0a0'.)  The docstring should say this, and the wrong (obsolete?) comment removed.

I would remove 'or not str(result)' below and consider raising a TypeError if the dialog somehow started returning a non-string.  But this is up to Serhiy.

Reply:
Tcl_Obj is the name of a Python object that wraps a Tcl's Tcl_Obj. Tkinter converts some Tcl objects to common Python types (int, str, tuple, etc), but can return a Tcl_Obj if Tcl uses a specialized type for some kind of objects. This can happen in every new version. The trick with 'or not str(result)' looks correct and may be necessary. The comment is correct and accurately describes the situation.

Reply:
So I learn something new ;-)  The only documentation is one reference in tkinter.__init__: `_tkinter.Tcl_Obj`.  Perhaps `_tkinter` should be added here.